src/manager/node.c

```diff
@@ -1511,6 +1513,50 @@ static int node_method_reload_unit(sd_bus_message *m, void *userdata, UNUSED sd_
         return node_run_unit_lifecycle_method(m, (Node *) userdata, "reload", "ReloadUnit");
 }
 
+/*************************************************************************
+ ********** org.containers.hirte.Node.SetLogLevel *******************
+ ************************************************************************/
+
+static int node_method_set_log_level(sd_bus_message *m, UNUSED void *userdata, UNUSED sd_bus_error *ret_error) {
+        const char *level = NULL;
+        Node *node = (Node *) userdata;
+        sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *sub_m = NULL;
+
+        int r = sd_bus_message_read(m, "s", &level);
+        if (r < 0) {
+                hirte_log_errorf("Failed to set log level call: %s", error.message);
+                r = sd_bus_reply_method_return(m, "");
+                if (r < 0) {
+                        hirte_log_errorf("Failed to send reply to proxy request: %s", strerror(-r));
+                }
+        }
+        LogLevel loglevel = string_to_log_level(level);
+        if (loglevel == LOG_LEVEL_INVALID) {
+                hirte_log_error("Failed to set log level call: Incorrect loglevel name");
+                r = sd_bus_reply_method_return(m, "");
+                if (r < 0) {
+                        hirte_log_errorf("Failed to send reply to proxy request: %s", strerror(-r));
+                }
+        }
+        r = sd_bus_call_method(
+                        node->agent_bus,
+                        HIRTE_AGENT_DBUS_NAME,
+                        INTERNAL_AGENT_OBJECT_PATH,
+                        INTERNAL_AGENT_INTERFACE,
+                        "SetLogLevel",
+                        &error,
+                        &sub_m,
+                        "s",
+                        level);
+        if (r < 0) {
+                hirte_log_errorf("Failed to set log level call: %s", error.message);
+                sd_bus_error_free(&error);
+                return false;
```

Comment:
```suggestion
                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
```
In case the call to the agent fails we need to forward an error to the caller. You can also add more detail to the message, e.g. instead of just `Internal error` you could state `Setting log level call to agent failed`.